server/chat-plugins/scavengers.js

```diff
@@ -216,6 +222,69 @@ function formatOrder(place) {
 	return place + 'th';
 }
 
+class ScavengerHuntDatabase {
+	static getRecycledHuntFromDatabase() {
+		// Return a random hunt from the database.
+		return scavengersData.recycledHunts[Math.floor(Math.random() * scavengersData.recycledHunts.length)];
+	}
+
+	static addRecycledHuntToDatabase(hosts, params) {
+		const huntSchema = {
+			hosts: hosts,
+			questions: [],
+		};
+
+		let questionSchema = {
+			text: '',
+			answers: [],
+			hints: [],
+		};
+
+		for (let i = 0; i < params.length; ++i) {
+			if (i % 2 === 0) {
+				questionSchema.text = params[i];
+			} else {
+				questionSchema.answers = params[i];
+				huntSchema.questions.push(questionSchema);
+				questionSchema = {
+					text: '',
+					answers: [],
+					hints: [],
+				};
+			}
+		}
+
+		scavengersData.recycledHunts.push(huntSchema);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static removeRecycledHuntFromDatabase(index) {
+		scavengersData.recycledHunts.splice(index - 1, 1);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static addHintToRecycledHunt(huntNumber, questionNumber, hint) {
+		scavengersData.recycledHunts[huntNumber - 1].questions[questionNumber - 1].hints.push(hint);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static removeHintToRecycledHunt(huntNumber, questionNumber, hintNumber) {
+		scavengersData.recycledHunts[huntNumber - 1].questions[questionNumber - 1].hints.splice(hintNumber - 1);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
```

Comment:
All of `FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));` probably should be its own function.